assembly/src/parsers/io_ops/env_ops.rs

```diff
@@ -1,51 +1,59 @@
-use super::{
-    parse_u32_param, push_value, validate_operation, AssemblyError, Felt, Operation, Token, Vec,
-};
+use super::{parse_u32_param, push_value, AssemblyError, Felt, Operation, Token, Vec};
 
 // ENVIRONMENT INPUTS
 // ================================================================================================
 
-/// Appends machine operations to the current span block according to the requested environment
-/// assembly instruction.
-///
-/// - `push.env.locaddr.i` pushes the absolute address of the local variable at index `i` onto the
-/// stack.
-/// - `push.env.sdepth` pushes the current depth of the stack onto the top of the stack, which is
-/// handled directly by the `SDEPTH` operation.
+/// Appends `locaddr.i` operation to the span block to push the absolute address of the local
+/// variable at index `i` onto the stack.
 ///
 /// # Errors
 ///
-/// This function expects a valid assembly environment op that specifies the environment input to
-/// be handled. It will return an error if the assembly instruction is malformed or the environment
-/// input is unrecognized.
-pub fn parse_push_env(
+/// It will return an error if the assembly instruction is malformed or it has inappropriate
+/// parameter value according to the number of local variables of the procedurethe
+pub fn parse_locaddr(
     span_ops: &mut Vec<Operation>,
     op: &Token,
     num_proc_locals: u32,
 ) -> Result<(), AssemblyError> {
-    validate_operation!(op, "push.env.locaddr|sdepth");
-
-    // update the span block
-    match op.parts()[2] {
-        "locaddr" => {
-            if num_proc_locals == 0 {
-                return Err(AssemblyError::invalid_op_with_reason(
-                    op,
-                    "no procedure locals available in current context",
-                ));
-            }
-            validate_operation!(@only_params op, "push.env.locaddr", 1);
-            let index = parse_u32_param(op, 3, 0, num_proc_locals - 1)?;
-
-            push_value(span_ops, -Felt::new(index as u64));
-            span_ops.push(Operation::FmpAdd);
-        }
-        "sdepth" => {
-            validate_operation!(@only_params op, "push.env.sdepth", 0);
-            span_ops.push(Operation::SDepth);
-        }
-        _ => return Err(AssemblyError::invalid_op(op)),
+    if op.parts()[0] != "locaddr" {
+        return Err(AssemblyError::unexpected_token(op, "locaddr"));
+    }
+    if op.num_parts() < 2 {
+        return Err(AssemblyError::missing_param(op));
+    }
+    if op.num_parts() > 2 {
+        return Err(AssemblyError::extra_param(op));
+    }
```

Comment:
nit: maybe a more compact way to do this is to combine it with index parsing. Maybe something like this:
```Rust
let index = match op.num_parts() {
    0 | 1 => return Err(AssemblyError::missing_param(op)),
    2 => parse_u32_param(op, 1, 0, num_proc_locals - 1)?,
    _ => return Err(AssemblyError::extra_param(op)),
};
```
If we go with this approach, we'll need to move the check for `num_proc_locals` above this block.